### PR TITLE
[Inductor UT] Generalize inductor UT for intel GPU (Part 3)

### PR DESCRIPTION
**Pull request by hoshibara**
[Inductor UT] Reuse Inductor test case for Intel GPU.
Reuse `test/inductor/test_compiled_autograd.py`

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @rec

### CODE REVIEW

File: test/inductor/test_compiled_autograd.py [hunk 1]
```diff
@@ -810,9 +810,12 @@ def fn():
 
         self.check_output_and_recompiles(fn, count=1)
 
-    @unittest.skipIf(not HAS_CUDA, "requires cuda")
+    @unittest.skipIf(not HAS_GPU, "requires gpu")
     def test_issue106555(self):
-        DEVICE = torch.device("cuda:0")
+        if HAS_CUDA:
```
Comment (etaf) [inline test/inductor/test_compiled_autograd.py:815]: suggest:
```
DEVICE = torch.device(GPU_TYPE, 0)
```

File: test/inductor/test_compiled_autograd.py [hunk 2]
```diff
@@ -855,7 +858,10 @@ def _forward(self, x):
                 x = x + self.module_with_jit_2(x.transpose(-2, -3)).transpose(-2, -3)
                 return x
 
-        torch.cuda.set_device(device=DEVICE)
+        if HAS_CUDA:
```
Comment (etaf) [inline test/inductor/test_compiled_autograd.py:861]: Suggest to use general device interface.
eg:
```
device_interface = get_interface_for_device(GPU_TYPE)
device_interface.set_device(device=DEVICE)
```

File: test/inductor/test_compiled_autograd.py [hunk 3]
```diff
@@ -2091,17 +2097,23 @@ def fn():
...
+    @unittest.skipIf(not HAS_GPU, "requires gpu")
     def test_free_activation_memory(self):
         script = """
 import torch
+from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_CUDA, HAS_XPU
 
 def main():
-    assert(torch.cuda.memory_allocated() == 0)
+    device_module = None
+    if HAS_CUDA:
```
Comment (etaf) [inline test/inductor/test_compiled_autograd.py:2108]: please extend the general device_interface in https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/device_interface.py
and use `device_interface.memory_allocated`

File: test/inductor/test_compiled_autograd.py [hunk 4]
```diff
@@ -20,11 +20,12 @@
 from torch import _inductor as inductor
 from torch._dynamo import compiled_autograd, config
 from torch._dynamo.backends.debugging import aot_eager
+from torch._dynamo.device_interface import get_interface_for_device
 from torch._dynamo.utils import counters
 from torch._inductor import config as inductor_config
 from torch._inductor.test_case import run_tests, TestCase
 from torch.testing._internal.common_utils import skipIfWindows
-from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
+from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_CPU, HAS_CUDA, HAS_GPU
```
Comment (etaf) [inline test/inductor/test_compiled_autograd.py:28]: Do we still need HAS_CUDA?

Comment (hoshibara) [reply test/inductor/test_compiled_autograd.py:28]: There are some cases for cudagraphs, they are checked by HAS_CUDA.

**Review by etaf — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**